android_emulator_cli.sh

```diff
@@ -93,46 +128,121 @@ fi
-echo "Downloading Android Platform Tools and Emulator files."
-yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager "build-tools;29.0.3"
+
+sleep .5
+echo "Downloading an Android API $ANDROID_API System Imag for $CPU_ABI."
+SYSTEM_IMAGE="system-images;android-$ANDROID_API;google_apis_playstore;$CPU_ABI"
+yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager $SYSTEM_IMAGE
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 echo ""
-echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 echo ""
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+sleep .5
+
+echo "Downloading Android Platform Tools and Emulator files."
+
+BUILD_TOOLS=$(~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager --list | grep "build-tools"  | tail -1 | cut -f1 -d'|' | sed 's: ::g' )
+
+echo "Build tools found: $BUILD_TOOLS"
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager "$BUILD_TOOLS"
+
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+echo ""
+echo ""
 echo "Updating emulator files"
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
 yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager --update
+
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+echo ""
+
+sleep .5
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
-echo "Creating a new Android Emulator."
+# Need to accept licenses for any installed packages 
+# https://developer.android.com/studio/command-line/sdkmanager#accept-licenses
+# If already accepted, then will skip.
+echo "Checking if any Licenses need to be accepted"
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+yes | ~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager --licenses
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
-echo "no" | ~/Library/Android/sdk/cmdline-tools/latest/bin/avdmanager create avd -n PixelXL29 -d "pixel_xl" -k "system-images;android-29;google_apis_playstore;x86_64"
 echo ""
+
+sleep .5
+
+# SDKManager Platform-Tools & Emulator END
+########################################################
+# Android Emulator creation & launch START
+
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
-echo "Setting emulator hardware settings."
+echo; echo;
+EMULATOR_NAME="PixelXL_"$ANDROID_API"_"$CPU_ABI""
+echo "Creating a new Android Emulator called $EMULATOR_NAME."
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 
+echo "no" | ~/Library/Android/sdk/cmdline-tools/latest/bin/avdmanager create avd -n $EMULATOR_NAME -d "pixel_xl" -k $SYSTEM_IMAGE --force
+
+echo ""
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+echo "Setting emulator hardware settings."
+echo ""
+
+sleep .5
+
+echo ""
+echo "no" | ~/Library/Android/sdk/cmdline-tools/latest/bin/avdmanager create avd -n ${EMULATOR_NAME} -d "pixel_xl" -k ${SYSTEM_IMAGE} --force
+ 
+sleep 2
```

Comment:
Currently keeping the `pixel xl` as default emulator "skin", but will allow choices in an upcoming PR.

Reply:
:nice: